src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -444,6 +578,87 @@ static void Test(params IEnumerable a)
                 );
         }
 
+        [Fact]
+        public void LanguageVersion_01_Declaration()
+        {
+            var src = @"
+class Program
+{
+    static void Test1(params System.ReadOnlySpan<long> a) {}
+    static void Test2(params long[] a) {}
+
+    void Test()
+    {
+        Test1(1);
+        Test2(2);
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll, parseOptions: TestOptions.RegularNext);
+            comp.VerifyDiagnostics();
+
+            comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll, parseOptions: TestOptions.Regular12);
+            comp.VerifyDiagnostics(
+                // (4,23): error CS8652: The feature 'params collections' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     static void Test1(params System.ReadOnlySpan<long> a) {}
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "params System.ReadOnlySpan<long> a").WithArguments("params collections").WithLocation(4, 23)
+                );
+        }
+
+        [Fact]
+        public void LanguageVersion_02_CallSite()
+        {
+            var src1 = @"
+public class Params
+{
+    static public void Test1(params System.ReadOnlySpan<long> a) {}
```

Comment:
Do we carry params-ness through to delegate natural types? Consider adding some variants of this test for delegate conversions as well, even if we don't, just to verify that behavior.

Reply:
> Do we carry params-ness through to delegate natural types? 

At the moment this is done only for arrays. Expanding the behavior to other types is on my list to do.

> Consider adding some variants of this test for delegate conversions as well,

I prefer to do that once "carry-over" part is implemented. But this is definitely a good scenario to cover.

Reply:
> At the moment this is done only for arrays. Expanding the behavior to other types is on my list to do.

Gotcha, thanks.